inc/Modules/Settings/Settings.php

```diff
@@ -0,0 +1,354 @@
+	 *
+	 * @param mixed $old_value The old value.
+	 * @param mixed $new_value The new value.
+	 */
+	public function on_site_type_change( $old_value, $new_value ): void {
+		if ( self::SITE_TYPE_CONSUMER !== $new_value ) {
+			return;
+		}
+
+		// By getting the API key, it will be generated if it doesn't exist.
+		self::get_api_key();
+	}
+
+	/**
+	 * Sanitize the `shared_sites` option.
+	 *
+	 * @param mixed $input The input value.
+	 *
+	 * @return array{
+	 * id: string,
+	 * name: string,
+	 * url: string,
+	 * api_key: string
+	 * }[]
+	 */
+	public static function sanitize_shared_sites( $input ): array {
+		if ( ! is_array( $input ) || empty( $input ) ) {
+			return [];
+		}
+
+		$sanitized = [];
+
+		foreach ( $input as $site_data ) {
+			if ( ! is_array( $site_data ) ) {
+				continue;
+			}
+
+			$site_id      = isset( $site_data['id'] ) ? sanitize_text_field( $site_data['id'] ) : '';
+			$site_name    = isset( $site_data['name'] ) ? sanitize_text_field( $site_data['name'] ) : '';
+			$site_url     = isset( $site_data['url'] ) ? esc_url_raw( $site_data['url'] ) : '';
+			$site_api_key = isset( $site_data['api_key'] ) ? sanitize_text_field( $site_data['api_key'] ) : '';
+
+			// Only save if required fields are filled.
+			if ( empty( $site_name ) || empty( $site_url ) ) {
+				continue;
+			}
+
+			$sanitized[] = [
+				'id'      => $site_id ?: wp_generate_uuid4(),
+				'name'    => $site_name,
+				'url'     => untrailingslashit( $site_url ),
+				'api_key' => $site_api_key,
+			];
+		}
+
+		return $sanitized;
+	}
+
+	/**
+	 * Static setters and getters for the individual settings.
+	 */
+
+	/**
+	 * Get brand sites configured for this governing site.
+	 *
+	 * @return array<string,array{
+	 *  api_key: string,
+	 *  id: string,
+	 *  name: string,
+	 *  url: string,
+	 * }>
+	 */
+	public static function get_shared_sites(): array {
+		$brands = get_option( self::OPTION_GOVERNING_SHARED_SITES, null ) ?: [];
+
+		$brands_to_return = [];
+		foreach ( $brands as $brand ) {
+			if ( ! is_array( $brand ) ) {
+				continue;
+			}
+
+			$brands_to_return[ $brand['url'] ] = [
+				'api_key' => $brand['api_key'] ?? '',
+				'id'      => $brand['id'] ?? '',
+				'name'    => $brand['name'] ?? '',
+				'url'     => $brand['url'] ?? '',
+			];
+		}
+
+		return $brands_to_return;
+	}
+
+	/**
+	 * Get a single brand site by URL
+	 *
+	 * @param string $site_url The site URL.
+	 *
+	 * @return ?array{
+	 *   api_key: string,
+	 *   id: string,
+	 *   name: string,
+	 *   url: string,
+	 * }
+	 */
+	public static function get_shared_site_by_url( string $site_url ): ?array {
+		$brand_sites = self::get_shared_sites();
+
+		$normalized_url = trailingslashit( $site_url );
+
+		return $brand_sites[ $normalized_url ] ?? null;
+	}
+
+	/**
+	 * Set the shared sites.
+	 *
+	 * @param array<string,array<string,mixed>> $sites The sites to set.
+	 *
+	 * @phpstan-param array<string,array{
+	 *   api_key?: string,
```

Comment:
We also need to be encrypting/decrpyting our API key here. See the latest OneSearch/OneUpdate